types/simulation/types.go

```diff
@@ -95,11 +98,20 @@ func NewOperationMsg(msg sdk.Msg, ok bool, comment string, cdc *codec.ProtoCodec
 		moduleName = msgType
 	}
 
-	if legacyMsg, okType := msg.(legacytx.LegacyMsg); okType {
-		return NewOperationMsgBasic(moduleName, msgType, comment, ok, legacyMsg.GetSignBytes())
+	anyMsg, err := types.NewAnyWithValue(msg)
+	if err != nil {
+		panic(err)
+	}
+	encoder := aminojson.NewAminoJSON()
```

Comment:
```suggestion
	encoder := aminojson.NewEncoder()
```

The constructor name should match the type name

Reply:
Agreed. This should be a separate PR since it necessitates an x/tx release, please review when you get a chance.
https://github.com/cosmos/cosmos-sdk/pull/16044